codegen/templates/tchannel_endpoint.tmpl

```diff
@@ -134,11 +136,13 @@ func (h *{{$handlerName}}) Handle(
 	{{end}}
 
 	{{- if .ResHeaders}}
-	if err := wfResHeaders.Ensure({{.ResHeaders | printf "%#v" }}, h.endpoint.Logger); err != nil {
-		return false, nil, nil, errors.Wrapf(
-			err, "%s.%s (%s) missing response headers",
-			h.endpoint.EndpointID, h.endpoint.HandlerID, h.endpoint.Method,
-		)
+	if wfResHeaders != nil {
```

Comment:
This is not correct. `Ensure` checks if the header contain certain keys and return error if not. If the header is nil, it should return error instead of skipping the check.